packages/runtime/runtime-definitions/src/protocol.ts

```diff
@@ -65,3 +69,38 @@ export interface IAttachMessage {
 export type InboundAttachMessage = Omit<IAttachMessage, "snapshot"> & {
 	snapshot: IAttachMessage["snapshot"] | null;
 };
+
+/**
+ * This is the message type that is used within the runtime when processing a sequenced message.
+ * It is the same as ISequencedDocumentMessage, but without the contents and clientSequenceNumbers
+ * which are sent separately. The contents are modified at multiple layers in the stack so having it
+ * separate doesn't require packing and unpacking the entire message.
+ * @alpha
+ * @legacy
+ */
+export type ISequencedMessageEnvelope = Omit<
+	ISequencedDocumentMessage,
+	"contents" | "clientSequenceNumber"
+>;
+
+/**
+ * These are the contents of a runtime message as it is processed throughout the stack.
+ * @alpha
+ * @legacy
+ */
+export interface IRuntimeMessagesContent {
+	contents: unknown;
+	localOpMetadata: unknown;
+	clientSequenceNumber: number;
+}
+
+/**
+ * The properties for the processMessages function in the runtime.
+ * @alpha
+ * @legacy
+ */
+export interface IProcessMessagesProps {
+	message: ISequencedMessageEnvelope;
```

Comment:
I note about naming. It matters :)

I think this should be `envelope`.
When I see in the code `props.message.type`, it's a bit hard to say (just looking at it without digging) if it's a single message or not, what props.message refers to.
I think props -> collection or messages, message -> envelope
That way it's clear that a collection has an envelope and the later has a type. And it's one envelope per collection (one can get it without looking at the types)